e2e/tests/app.e2e.ts

```diff
@@ -0,0 +1,148 @@
+import { expect, test } from '@playwright/test';
+import { AngularJSPanelPage } from '../pages/angularjs-panel-page';
+import { AngularPanelPage } from '../pages/angular-panel-page';
+import { AngularjsMaterialTabsPage } from '../pages/angularjs-material-tabs-page';
+import { AngularjsMaterialButtonsPage } from '../pages/angularjs-material-buttons-page';
+import { AngularMaterialTabsPage } from '../pages/angular-material-tabs-page';
+import { AngularMaterialCdkTreePage } from '../pages/angular-material-cdk-tree-page';
+import { AngularMaterialButtonsPage } from '../pages/angular-material-buttons-page';
+import { TopNavPage } from '../pages/top-nav-page';
+
+test.describe('AngularJS-Angular hybrid app', () => {
+  test.beforeEach(async ({page}) => {
+    await page.goto('');
+  });
+
+  test.afterEach(async ({ page }) => {
+    const errorLogs: string[] = [];
+    page.on('console', message => {
+      if (message.type() === 'error') {
+        errorLogs.push(message.text());
+      }
+    });
```

Comment:
Registering the listener after the test will not capture any messages.
Also, this will only capture `console.error()` logs. It would be good to also capture uncaught thrown errors. We can do this with the [`pageerror` event][1].

You need to register the listener before the test. For example, something like:

```ts
test.describe('AngularJS-Angular hybrid app', () => {
  let errorLogs: string[];

  test.beforeEach(async ({page}) => {
    errorLogs = [];

    page.on('console', message => {
      if (message.type() === 'error') {
        errorLogs.push(message.text());
      }
    });
    page.on('pageerror', err => {
      errorLogs.push(err.message);
    });

    await page.goto('');
  });

  test.afterEach(() => {
    expect(errorLogs).toStrictEqual([]);
  });
```

Alternatively, you can override `page` to automatically register the listeners before running each test as discussed [here][2].

##
BTW, this applies to other similar PRs that add e2e tests with Playwright.

[1]: https://playwright.dev/docs/api/class-page#page-event-page-error
[2]: https://github.com/microsoft/playwright/discussions/11690#discussioncomment-2060397

Reply:
💯 Good catch!

Changed